### PR TITLE
feat: add headers in kafka event source

### PR DESCRIPTION
**Pull request by cmontois**
Goal: fixing https://github.com/argoproj/argo-events/issues/2442.

First open source PR (and first GO PR) :)

### CODE REVIEW

File: eventsources/sources/kafka/start.go [hunk 1]
```diff
@@ -223,6 +223,15 @@ func (el *EventListener) partitionConsumer(ctx context.Context, log *zap.Sugared
...
 			Metadata:  kafkaEventSource.Metadata,
 		}
+
+		headers := make(map[string]string)
+
+		for _, recordHeader := range msg.Headers {
+			headers[string(recordHeader.Key)] = string(recordHeader.Value)
+		}
+
+		eventData.Headers = headers
```
Comment (whynowy) [inline eventsources/sources/kafka/start.go:233]: Thanks, could you also add it to `consumerGroupConsumer()`?

**Comment by whynowy** [comment]
> Goal: fixing #2442.
> 
> First open source PR (and first GO PR) :)

So honored!

File: eventsources/sources/kafka/start.go (hunk 1 above)
Comment (cmontois) [reply eventsources/sources/kafka/start.go:233]: That's should do it !

**Review by whynowy — APPROVED** [review]
LGTM, thanks!

**Comment by whynowy** [comment]
@cmontois - can you run `go mod tidy` and push it again?

**Comment by cmontois** [comment]
> 



> @cmontois - can you run `go mod tidy` and push it again?

@whynowy done

**Comment by OneCricketeer** [comment]
Should we close #2442 ?

**Pull request merged**